src/main/java/org/humancellatlas/ingest/schemas/web/SchemaSearchProcessor.java

```diff
@@ -0,0 +1,34 @@
+package org.humancellatlas.ingest.schemas.web;
+
+
+import org.humancellatlas.ingest.schemas.Schema;
+
+import org.springframework.data.rest.webmvc.RepositorySearchesResource;
+import org.springframework.hateoas.ResourceProcessor;
+import org.springframework.stereotype.Component;
+
+
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.*;
+
+
+/**
+ * Created by rolando on 23/04/2018.
+ */
+@Component
+public class SchemaSearchProcessor implements ResourceProcessor<RepositorySearchesResource> {
+
+    @Override
+    public RepositorySearchesResource process(RepositorySearchesResource searchesResource) {
+        if(searchesResource.getDomainType().equals(Schema.class)) {
+            searchesResource.add(linkTo(methodOn(SchemaController.class).querySchemas(null,
```

Comment:
Encapsulate `null, null, null, null, null` into a simpler method say, `querySchemas()`, which implies all null arguments. This way we just call `querySchemas()` without the reader having to figure out what those `null`s are.

Reply:
That won't work unfortunately, have to insert all the paramaters as the LinkBuilder uses them to generate the templated URL